packages/core/src/app/checkout/getCheckoutStepStatuses.ts

```diff
@@ -34,7 +34,7 @@ const getCustomerStepStatus = createSelector(
         const isComplete = hasEmail || isUsingWallet;
         const isEditable = isComplete && !isUsingWallet && isGuest
 
-        if (config?.checkoutSettings.providerWithCustomCheckout === PaymentMethodId.StripeUPE && hasEmail && isGuest) {
+        if (!isUsingWallet && config?.checkoutSettings.providerWithCustomCheckout === PaymentMethodId.StripeUPE && hasEmail && isGuest) {
```

Comment:
Can we please add tests to cover this? Also can we work on this control flow, feels like we are just adding conditions one after the other. 

Have you tested this with other wallet buttons?

Reply:
Hello @animesh1987, I already added unit testing, also, QA already validated the wallets available for StripeUPE, I added a video of the missing wallet https://drive.google.com/file/d/1N7JxCWImcA960SqjC6eaX1N9zLEmBb8u/view?usp=sharing, and about the approach let me explain a little each validation inside the If validation for stripeLink.

The validation is the following:
`if (!isUsingWallet && config?.checkoutSettings.providerWithCustomCheckout === PaymentMethodId.StripeUPE && hasEmail && isGuest)`

**`config?.checkoutSettings.providerWithCustomCheckout === PaymentMethodId.StripeUPE`**: this validation was suggested by you, it comes from BCAPP and tells us when StripeLink is active.

**`hasEmail`**: this validation was added to make sure that the flow is only used if an email has already been saved, it could be deleted but we prefer to exclude it from the custom flow when it is not necessary.

**`isGuest`**: Stripelink only works with guest so in the case of using the BC customer flow we are not interested in having the custom flow.

and the last one, **`isUsingWallet`** : which is used to exclude wallets from the custom flow, since they don't need to reload stripe components in case of a reload.

please let me know any concern :D

Reply:
Because this is part of the core logic of checkout-js, `!isUsingWallet && config?.checkoutSettings.providerWithCustomCheckout === PaymentMethodId.StripeUPE && hasEmail && isGuest` is simply too complex for future developers to comprehend.

Could we replace this long line with a function like `isUsingStripeLink()`? (This may not be the best name. please summarise the conditions' purpose.)

Reply:
Hey Alan thanks for the explanation, it does feel a bit too many conditions still. 

On the condition `config?.checkoutSettings.providerWithCustomCheckout === PaymentMethodId.StripeUPE` what happens when we add another accelerated checkout in future?


On another note in the video above, I can see wallet buttons still active. Thought there was a decision made somewhere to remove wallet buttons if stripe link was active?

Reply:
Hello @bc-peng, as we saw by message, I made a constant instead of a function so to avoid sending many parameters, let me know if It's ok and if not I can change it for a function :D, I also added a comment explaining the reason for the `constant` and the `if` 

In response to your questions @animesh1987,
Apparently, there can only be one accelerated checkout working. 
For example:
**Bolt** and **StripeLink** cannot coexist. I understand that the **Bolt team** is working on these validations, so if we add another accelerated checkout in the future, there will be a standard flow to avoid **StripeLink** as well as **Bolt**.

And in response to your last question, It's partially correct, we remove the wallets for **StripeLink** but as long as the user is authenticated in **Stripe** (In the demo, I am still a guest for both **BC** and **Stripe**)

Let me know any other questions, regards